tests/protocols/test_websocket.py

```diff
@@ -52,15 +52,17 @@ def run_loop(loop):
 
 
 @contextmanager
-def run_server(app, protocol_cls):
+def run_server(app, protocol_cls, path="/"):
     asyncio.set_event_loop(None)
     loop = asyncio.new_event_loop()
     config = Config(app=app, ws=protocol_cls)
     server_state = ServerState()
     protocol = functools.partial(H11Protocol, config=config, server_state=server_state)
     create_server_task = loop.create_server(protocol, host="127.0.0.1")
     server = loop.run_until_complete(create_server_task)
-    url = "ws://127.0.0.1:%d/" % server.sockets[0].getsockname()[1]
+    url = "ws://127.0.0.1:{port}{path}".format(
+        port=server.sockets[0].getsockname()[1], path=path
```

Comment:
I guess we may as well pull `port = server.sockets[0].getsockname()[1]` out to the line above, and then use `.format(port=port, path=path)`

Reply:
Done: https://github.com/encode/uvicorn/pull/372/commits/4b61f2fbd97a469a25b790a3eb706f923fe9bcfe